library/ssl_tls13_server.c

```diff
@@ -3146,6 +3156,16 @@ static int ssl_tls13_prepare_new_session_ticket(mbedtls_ssl_context *ssl,
 
     MBEDTLS_SSL_PRINT_TICKET_FLAGS(4, session->ticket_flags);
 
+#if defined(MBEDTLS_SSL_EARLY_DATA) && defined(MBEDTLS_SSL_ALPN)
+    if (ssl->alpn_chosen != NULL) {
+        session->alpn = mbedtls_calloc(strlen(ssl->alpn_chosen) + 1, sizeof(char));
```

Comment:
We can send several tickets within the same session thus this will leak memory. We probably just have to check if `session->ticket_alpn != NULL`. Otherwise, it would probably be beneficial to introduce a function similar to ` mbedtls_ssl_session_set_hostname` for the ticket ALPN: `mbedtls_ssl_session_set_ticket_alpn()`? We would then use it here and in `session_load`.